src/main/java/com/linecorp/armeria/client/http/HttpRequestSubscriber.java

```diff
@@ -135,8 +139,23 @@ private void writeFirstHeader() {
                          host, firstHeaders.method().name(), firstHeaders.path());
         logBuilder.attr(RequestLog.HTTP_HEADERS).set(firstHeaders);
 
-        encoder.writeHeaders(ctx, id, streamId(), firstHeaders, false).addListener(this);
-        ctx.flush();
+        switch (firstHeaders.method()) {
+            case GET:
+            case HEAD:
+            case PATCH:
+            case DELETE:
+            case TRACE:
+            case CONNECT:
+                // Do not send any content for the methods which is not supposed to have content.
+                // To be precise, the RFC 2616 does not prohibit sending non-empty content for these methods,
+                // but it is discouraged to do so.
+                // See: http://stackoverflow.com/q/978061
```

Comment:
To clarify, I think the correct behavior is don't set these headers if there is no content, not blocking the content completely.